src/vmm/LibVirtDriverKVM.cc

```diff
@@ -98,66 +90,52 @@ static void do_network_hosts(ofstream& file,
 /* -------------------------------------------------------------------------- */
 /* -------------------------------------------------------------------------- */
 
-static int to_i(const string& sval)
+static int to_int(const string& s)
 {
-    int ival;
+    int val;
 
-    istringstream iss(sval);
+    istringstream iss(s);
 
-    iss >> ival;
+    iss >> val;
 
     if (iss.fail() || !iss.eof())
     {
         return -1;
     }
 
-    return ival;
+    return val;
 }
 
+template<typename T>
 static void insert_sec(ofstream& file, const string& base, const string& s,
                        const string& sm, const string& sml)
 {
-    int s_i = 0;
+    T s_i = 0;
 
     if (!s.empty())
     {
-        s_i = to_i(s);
-
-        if (s_i < 0)
-        {
-            return;
-        }
+        s_i = one_util::string_to_unsigned<T>(s);
 
-        file << "\t\t\t\t<" << base << "_sec>" << one_util::escape_xml(s)
+        file << "\t\t\t\t<" << base << "_sec>" << one_util::escape_xml(std::to_string(s_i))
              << "</" << base << "_sec>\n";
     }
 
     if (!sm.empty())
     {
-        int sm_i = to_i(sm);
+        const auto sm_i = one_util::string_to_unsigned<T>(sm);
 
-        if (sm_i < 0)
-        {
-            return;
-        }
-
-        if ( sm_i > s_i )
+        if ( sm_i > s_i)
         {
             file << "\t\t\t\t<" << base << "_sec_max>"
-                 << one_util::escape_xml(sm)
+                 << one_util::escape_xml(std::to_string(sm_i))
                  << "</" << base << "_sec_max>\n";
 
             if (!sml.empty())
             {
-                int sml_i = to_i(sml);
-
-                if (sml_i < 0)
-                {
-                    return;
-                }
+                const auto sml_i = one_util::string_to_unsigned<T>(sml);
 
                 file << "\t\t\t\t<" << base << "_sec_max_length>"
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Potential log injection detected. Ensure all untrusted input is properly sanitized before logging. Use parameterized logging or validate input against an allow list to prevent log injection vulnerabilities. Consider using a dedicated logging library&#39;s built-in sanitization features when available. Learn more - https://cwe.mitre.org/data/definitions/117.html